src/composables/use-cookies.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * This composable enables setting and getting app cookies.
+ * Normally, we cannot call the `app.$cookies.set` function from outside a
+ * `setup` function. Here, we overcome this constraint by passing the `app`
+ * parameter into the composable and thus saving the necessary context.
+ */
+import { isProd } from '~/utils/node-env'
+
+import type { OpenverseCookies } from '~/types/cookies'
+
+import type { NuxtAppOptions } from '@nuxt/types'
+import type { CookieSerializeOptions } from 'cookie'
+
+const cookieOptions: CookieSerializeOptions = {
+  path: '/',
+  sameSite: 'strict',
+  maxAge: 60 * 60 * 24 * 60,
+  secure: isProd,
+}
+
+export const useCookies = (app: NuxtAppOptions) => {
```

Comment:
Naïve question: why not use the [`useCookies` composable](https://vueuse.org/integrations/usecookies/#usecookies) from VueUse?

Reply:
Thank you fro asking this question, @dhruvkb !

I've started answering why we can't use it, and then found a much nicer solution.

We use the [`cookie-universal-nuxt`](https://github.com/microcipcip/cookie-universal/tree/master/packages/cookie-universal-nuxt) which injects `$cookies` to the Nuxt `app`. This means that we can only set cookies in components and not in a Pinia store because you can't access `app` from outside of a `setup` function.

That said, researching this issue, I've found that `$nuxt` [was added to the Pinia Nuxt plugin](https://github.com/vuejs/pinia/issues/186#issuecomment-810950917), so we can set the cookies from the store by calling `this.$nuxt.app.$cookies.set()`.